slackcli/names.py

```diff
@@ -28,6 +28,14 @@ def username(self, user_id):
             self.user_index[user_id] = slack.client().users.info(user_id).body['user']['name']
         return self.user_index[user_id]
 
+    def user_id(self, username):
+        members = slack.client().users.list().body['members']
+        member = [x for x in members if x['name'] == username][0]
+        if member:
+            return member['id']
+        else:
+            return None
```

Comment:
The problem with this method is that it makes a costly Slack API request (or more if there are multiple pages of results) for every `@username` mention in the message. Is there an API for searching users by name? If yes, then you want to use that endpoint. And you need to cache that call in case there are multiple mentions of the same user in the message. This is the same mechanism that is used with `self.user_index` in the `username` method.

Reply:
I don't think there is an endpoint for that yet. I will try to implement caching